File: FE/src/features/role/services/roleService.ts
import { apiClient } from '../../../services/apiClient'
import type { PaginatedResult } from '../../../types/api.types'
import type { CreateRoleRequest, RoleResponse, UpdateRoleRequest } from '../types/role.types'

export const roleService = {
  getAll: (page = 1, size = 10, sort = 'id,asc') => {
    const params = new URLSearchParams({
      page: String(page),
      size: String(size),
      sort,
    })
    return apiClient<PaginatedResult<RoleResponse>>(`/roles?${params}`)
  },

  getById: (id: number) => apiClient<RoleResponse>(`/roles/${id}`),

  create: (data: CreateRoleRequest) =>
    apiClient<RoleResponse>('/roles', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  update: (data: UpdateRoleRequest) =>
    apiClient<RoleResponse>('/roles', {
      method: 'PUT',
      body: JSON.stringify(data),
    }),

  delete: (id: number) => apiClient<null>(`/roles/${id}`, { method: 'DELETE' }),
}

File: BE/src/main/java/Manager_vnd/Manager/feature/role/RoleService.java
package Manager_vnd.Manager.feature.role;

import Manager_vnd.Manager.dto.PaginatedResult;
import Manager_vnd.Manager.feature.role.dto.CreateRoleRequest;
import Manager_vnd.Manager.feature.role.dto.RoleResponse;
import Manager_vnd.Manager.feature.role.dto.UpdateRoleRequest;

public interface RoleService {

    PaginatedResult<RoleResponse> getAll(int page, int size, String sort);

    RoleResponse getById(long id);

    RoleResponse create(CreateRoleRequest request);

    RoleResponse update(UpdateRoleRequest request);

    void delete(long id);
}
